src/org/rascalmpl/util/maven/Artifact.java

```diff
@@ -151,6 +150,14 @@ private void calculateClassPath(Scope forScope, HashSet<WithoutVersion> alreadyI
         }
     }
 
+    private Artifact createSystemArtifact(Dependency d) {
+        var path = d.getSystemPath();
+        if (path == null) {
+            return Artifact.unresolved(d.getCoordinate(), messages.writer());
```

Comment:
Hmm, is this the right messages we and to provide?

Shouldn't way say something like could not resolve system dependency?

Reply:
Probably. I am not sure what exception to throw though. The best option is probably an "UnresolvableModelException" as that is used in other places. We have add `throws UnresolvableModelException` to a number of methods if we decide to go that route.

Reply:
I mean, we write messages to the IListWriter, to note errors in resolving  dependencies etc.

Reply:
I solved it differently: the problem is now detected earlier (in `Dependency.build`) and the dependency ignored. An error messages was already added to the list of messages in that case.

Reply:
You are right. Dependencies are no longer filtered. Errors where already generated by Maven itself. We just had to make sure not to use the systemPath if it was not present.